### PR TITLE
Implemented Raw Data Export for Multiple Cells

### LINKED ISSUE #203: refactor export to csv button
The "export to csv" button on the dashboard only exports the data from one cell. When a user selects multiple cells, it should export separate csv files for each of the cells. Also when exporting data from the cells within a range, it should be the raw data. Currently it's based off of an endpoint that aggregates cell data by the hour.

### LINKED ISSUE #210: Export to CSV not working
## Describe the Bug

The current bug was introduced after merging #209, and it pertains to the export CSV functionality, which is broken due to the startTime and endTime parameters not being passed in the correct format. This leads to the server returning a 500 error.

## To Reproduce

To reproduce this issue, follow these steps:

1. Navigate to the dashboard and click the "EXPORT TO CSV" button.
2. Observe the server returning a 500 error in the console.

## Expected Behavior

The expected behavior is that the export CSV functionality successfully sends a request to the server with the correct startTime and endTime parameters. This should result in the server processing the request and returning the requested data in a CSV file without encountering any errors.

### PR DESCRIPTION
**Pull request by ay-bh**
Implemented enhancements to the "export to csv" functionality on the dashboard. With this update, users can now export raw data for multiple cells as separate CSV files, addressing the issue where previously only aggregated hourly data could be exported for a single cell, resolves #203.  

#### Changes Made:

1. Added a `resample` query parameter to the backend API endpoint `/api/cell/data/${cellId}`, allowing the API to handle raw data export when `resample=none` is specified. This parameter is used to determine whether to perform data resampling or to fetch raw data.

2. Updated the backend database models to handle the new `resample` logic.

3. Revised the `getCellData` schema to include the `resample` parameter, enabling front-end requests to specify the desired resampling strategy.

4. Altered the front-end logic in `DownloadBtn.jsx` to pass the `resample=none` parameter when initiating the data download, ensuring that raw data is exported.

#### Testing Performed:

Manual testing was conducted to ensure the new functionalities work as expected. This includes:

- REST API calls to the endpoint with various resample values to validate backend response.
- Frontend tests for exporting multiple cells' data with different resampling settings to ensure proper CSV file generation.

EDIT: It also closes #210

### CODE REVIEW

File: frontend/src/services/cell.js
Comment (aaron-wu1) [inline frontend/src/services/cell.js:1]: Can you also add the default resample value as hourly, if resample isn't passed in

**Review by aaron-wu1 — CHANGES_REQUESTED** [review]

File: frontend/src/services/cell.js
Comment (ay-bh) [reply frontend/src/services/cell.js:1]: Done

File: backend/api/database/schemas/get_cell_data_schema.py
[the remark this answers is not in the document]
Comment (ay-bh) [reply backend/api/database/schemas/get_cell_data_schema.py:1]: Done

**Review by aaron-wu1 — CHANGES_REQUESTED** [review]
marshmallow is already imported with ma so we could use ma.validate instead

**Comment by ay-bh** [comment]
> marshmallow is already imported with ma so we could use ma.validate instead
@aaron-wu1 We need to import `validate` directly to access Marshmallow's validation functions, which aren't available as attributes of the Marshmallow instance (`ma`).

**Review by aaron-wu1 — APPROVED** [review]

**Pull request merged**